wgsl/index.bs

```diff
@@ -2704,122 +2787,151 @@ sampler_comparison
-     Rgba8i
-  | BGRA8UNORM
-     Rgba8  ???
-  | BGRA8UNORM-SRGB
-     ???
-  | RGB10A2UNORM
-     Rgb10A2  -- Capability: StorageImageExtendedFormats
-  | RG11B10FLOAT
-     R11fG11fB10f  -- Capability: StorageImageExtendedFormats
-  | RG32UINT
-     Rg32ui  -- Capability: StorageImageExtendedFormats
-  | RG32SINT
-     Rg32i  -- Capability: StorageImageExtendedFormats
-  | RG32FLOAT
-     Rg32f  -- Capability: StorageImageExtendedFormats
-  | RGBA16UINT
-     Rgba16ui
-  | RGBA16SINT
-     Rgba16i
-  | RGBA16FLOAT
-     Rgba16f
-  | RGBA32UINT
-     Rgba32ui
-  | RGBA32SINT
-     Rgba32i
-  | RGBA32FLOAT
-     Rgba32f
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>texture_sampler_types</dfn> :
 
-</pre>
+    | [=syntax/sampler_type=]
+
+    | [=syntax/depth_texture_type=]
+
+    | [=syntax/sampled_texture_type=] [=syntax/less_than=] [=syntax/type_decl=] [=syntax/greater_than=]
+
+    | [=syntax/multisampled_texture_type=] [=syntax/less_than=] [=syntax/type_decl=] [=syntax/greater_than=]
+
+    | [=syntax/storage_texture_type=] [=syntax/less_than=] [=syntax/texel_format=] [=syntax/comma=] [=syntax/access_mode=] [=syntax/greater_than=]
+</div>
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>sampler_type</dfn> :
+
+    | [=syntax/sampler=]
+
+    | [=syntax/sampler_comparison=]
+</div>
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>sampled_texture_type</dfn> :
+
+    | [=syntax/texture_1d=]
+
+    | [=syntax/texture_2d=]
+
+    | [=syntax/texture_2d_array=]
+
+    | [=syntax/texture_3d=]
+
+    | [=syntax/texture_cube=]
+
+    | [=syntax/texture_cube_array=]
+</div>
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>multisampled_texture_type</dfn> :
+
+    | [=syntax/texture_multisampled_2d=]
+</div>
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>storage_texture_type</dfn> :
+
+    | [=syntax/texture_storage_1d=]
+
+    | [=syntax/texture_storage_2d=]
+
+    | [=syntax/texture_storage_2d_array=]
+
+    | [=syntax/texture_storage_3d=]
+</div>
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>depth_texture_type</dfn> :
+
+    | [=syntax/texture_depth_2d=]
+
+    | [=syntax/texture_depth_2d_array=]
+
+    | [=syntax/texture_depth_cube=]
+
+    | [=syntax/texture_depth_cube_array=]
+
+    | [=syntax/texture_depth_multisampled_2d=]
+</div>
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>texel_format</dfn> :
+
+    | [=syntax/r8unorm=]
+
+    | [=syntax/r8snorm=]
+
+    | [=syntax/r8uint=]
+
+    | [=syntax/r8sint=]
+
+    | [=syntax/r16uint=]
+
+    | [=syntax/r16sint=]
+
+    | [=syntax/r16float=]
+
+    | [=syntax/rg8unorm=]
+
+    | [=syntax/rg8snorm=]
+
+    | [=syntax/rg8uint=]
+
+    | [=syntax/rg8sint=]
+
+    | [=syntax/r32uint=]
+
+    | [=syntax/r32sint=]
+
+    | [=syntax/r32float=]
+
+    | [=syntax/rg16uint=]
+
+    | [=syntax/rg16sint=]
+
+    | [=syntax/rg16float=]
+
+    | [=syntax/rgba8unorm=]
+
+    | [=syntax/rgba8unorm_srgb=]
+
+    | [=syntax/rgba8snorm=]
+
```

Comment:
Do we need all these newlines?

Reply:
AFAIK, through a CSS rule, we should be able to make these whitespaces unnecessary. Still, it is a trade-off between allowing line-breaks for rules themselves (which would look very clean if the rule authors adopt it, I didn't include instances for the sake of fewer LoC changes in this PR). Also, not allowing multiple-line rules would be a regression from the previous authoring.